### DISCUSSION TITLE
How to create formula for macOS & linux?

### DISCUSSION THREAD

**Discussion by sgoudham** [Homebrew/discussions#3580] [Writing Formulae/Casks] [answered]
Hi, I'm new to writing formulas in Homebrew. 
I was wondering how you can support `brew install` for macOS & linux(brew)? 

I've tried various configurations, including setting multiple URLs for different platforms using `on_macos` and `on_linux`, but I can't figure it out. I've tried researching this question and coming up with nothing which leads me to believe that I fundamentally misunderstand something. 

I'd appreciate any tips on how to make it possible. [Here](https://github.com/catppuccin/homebrew-tap/blob/main/Formula/mdbook-catppuccin.rb) is my current formula for one of my tools. It currently only includes macOS as I couldn't figure out how to include a linux `.tar.gz` 

I should mention that when I tried to use the above syntax to use 2 different urls, I kept getting `macOS cannot include 'url'` and assumed it to be intended behaviour, is this correct? 

Thanks!

**Comment by SMillerDev** [comment]
You seem to use formula as casks. If you want to distribute precompiled binaries you should use a cask.

**Reply by sgoudham** [reply]
I thought casks were meant for homebrew GUI applications?

**Reply by SMillerDev** [reply]
Those are usually precompiled binaries.

**Reply by sgoudham** [reply]
Right okay. So formula should usually be built from source? Are there any examples of formula being built from source that work on macOS & linux?

**Reply by SMillerDev** [reply, marked as answer]
Almost all formulae in homebrew-core work for both platforms.
